web/react/components/user_settings/user_settings_advanced.jsx

```diff
@@ -13,21 +13,38 @@ export default class AdvancedSettingsDisplay extends React.Component {
 
         this.updateSection = this.updateSection.bind(this);
         this.updateSetting = this.updateSetting.bind(this);
+        this.toggleFeature = this.toggleFeature.bind(this);
+        this.saveEnabledFeatures = this.saveEnabledFeatures.bind(this);
         this.setupInitialState = this.setupInitialState.bind(this);
 
-        this.state = this.setupInitialState();
+        this.state = {settings: {}, preReleaseFeatures: []};
+        this.setupInitialState();
     }
 
     setupInitialState() {
-        const sendOnCtrlEnter = PreferenceStore.getPreference(
-            Constants.Preferences.CATEGORY_ADVANCED_SETTINGS,
-            'send_on_ctrl_enter',
-            {value: 'false'}
-        ).value;
-
-        return {
-            settings: {send_on_ctrl_enter: sendOnCtrlEnter}
+        const settings = {
+            send_on_ctrl_enter: PreferenceStore.getPreference(
+                Constants.Preferences.CATEGORY_ADVANCED_SETTINGS,
+                'send_on_ctrl_enter',
+                {value: 'false'}
+            ).value
         };
+
+        let enabledFeatures = 0;
+        PreferenceStore.getPreferences(Constants.Preferences.CATEGORY_ADVANCED_SETTINGS).forEach((setting) => {
+            if (setting.name.lastIndexOf(Constants.FeatureTogglePrefix) === 0) {
+                settings[setting.name] = setting.value;
+                if (setting.value === 'true') {
+                    enabledFeatures++;
+                }
+            }
+        });
+
+        Client.getAvailablePreReleaseFeatures((data) => {
+            this.setState({settings, preReleaseFeatures: data || [], enabledFeatures});
```

Comment:
@crspeller Eslint didn't shout at me :)
I guess the reason is because `setState` is called asynchronously in the callback function of `Client.getAvailablePreReleaseFeatures`

I even couldn't get it to work using `this.state`, I think this is because the state has already been initialized, once the callback has fired. 
Is this really the "wrong" way to do it? Because I couldn't think of another solution - have any hints for me :)?